breadth_and_depth_first_search.md

```diff
@@ -0,0 +1,65 @@
+### Notes
+
+### General
+
+while i did manage to do the prework and am confident in my final answers, i'm
+not confident about graphs in general. that is to say, given that there are so
+many applications of graphs, even implementing something 'basic' like dfs or bfs
+is quite daunting because of all the potential applications, constraints, and so
+on. for example, i've read through the word search ladder and the knights tour a
```

Comment:
The word search ladder and knights tour are both quite tricky applications, and involve a lot of nuance that's kind of unrelated to graphs.